### ISSUE TITLE
Allow dynamic templates

### ISSUE THREAD

**Issue by gjdev** [swimlane/angular2-data-table#58]
It's currently not possible to have "dynamic" cell templates. Once rendered, the cell-template doesn't change anymore. It also doesn't "see" the context of its host component. E.g. something like this is not possible currently (where isEditing() can be triggered by e.g. a toggle in the host component):

```
  <template #tmpl let-row="row" let-value="value">
    <input *ngIf="isEditing()" [value]="value"/>
    <span *ngIf="!isEditing()">{{value}}</span>
  </template>
```

**Comment by amcdnl** [comment]
Can you provide a demo?

**Comment by adaur** [comment]
Interesting, it's something I am currently tring to do. I have a Edit button, and once the edition is complete the data in the datatable doesn't update itself whereas the example here does: https://github.com/DavideViolante/Angular2-Express-Mongoose

If I refresh the page or I search for a term that does not exist and remove it, it seems to "recall" the data somehow and the new value is displayed.

![video20160817000617](https://cloud.githubusercontent.com/assets/959078/17717517/c2d85b68-640e-11e6-826b-440ce68b772e.gif)

**Comment by amcdnl** [comment]
You need to update your data in the results itself. See this: https://github.com/swimlane/angular-data-table/blob/master/demos/inline-editing.html#L100

Its for ng1 version but still should apply. Open to suggestions on better approaches.

**Comment by adaur** [comment]
Thanks for your answer.

Actually, I would like to manually trigger the update, it should not be that hard, right?

I have

```
this.http.put("/cat/"+cat._id, JSON.stringify(cat), this.options).subscribe(
			res => {
				this.isEditing = false;
				this.cat = cat;
				this.route.data.subscribe(data => {
					this.temp = [...data];
					this.rows.push(...data)
				});
				this.sendInfoMsg("item edited successfully.", "success");
			},
			error => console.log(error)
		);
```

but this duplicates all the rows instead of replacing them.  Adding `this.rows = [];` doesn't seem to help.

Sorry for the flood :)

**Comment by amcdnl** [comment]
Is this getting triggered? https://github.com/swimlane/angular2-data-table/blob/master/src/components/DataTable.ts#L99

**Comment by adaur** [comment]
Sorry, I'm really a JS newbie... How can I check it? I added a console.log above it and it did not seem to pass there but I am not sure if the ts code was not cached before.

**Comment by adaur** [comment]
Interestingly, if I do

```
submitEdit(row) {
		this.http.put("/cat/"+row._id, JSON.stringify(row), this.options).subscribe(
			res => {
				this.isEditing = false;
				this.row = row;
				console.log(this.rows); // <-- this returns all the rows, including the update I just did
				this.sendInfoMsg("item edited successfully.", "success");
			},
			error => console.log(error)
		);
	}
```

Since the rows are changed, there must be some kind of even not being emitted properly.

**Comment by adaur** [comment]
I have created a simple function:

```
updateRows() {
		console.log('rows updated');
	}
```

then added this in the datatable DOM

```
<datatable
				class='material'
				[rows]='rows'
				[options]='optionsTable'
				(onRowsUpdate)='updateRows(
```)'>

But the event does not seem to get fired when I edit an element. It does at loading, though.

All I need is a reset() function to force the table to reload, or edit a specific row. Any hint on this?

**Comment by amcdnl** [comment]
Splice out the rows and re-insert them.

**Comment by adaur** [comment]
That's what I did later, copying the updateFilter function without the filter part, but it just does not work (ie: table not refreshing). I even tried to add an Update button to reload the rows... same result.

**Comment by amcdnl** [comment]
Can you make a plunkr demo?

**Comment by adaur** [comment]
It seems to work better than yesterday night, I'll investigate.

Still, after one click it should display the first entry only. You have to click several times to make them all go away.

https://plnkr.co/qoJk25kT0o8MbXistXjE

**Comment by fedotxxl** [comment]
+1 for this request. I've got custom template for rows selection/deselection. It calls component functions to get state (is row selected or not). When I change selection it doesn't refresh templates. How can I force it? 

note: I update `selected` array but it doesn't help...

**Comment by amcdnl** [comment]
Fixed in 0.3.5 :)

**Comment by vkurakula** [comment]
how to over ride default message of ngx-table like


 @Input() messages: any = {
    // Message to show when array is presented
    // but contains no values
    emptyMessage: 'No data to display',

    // Footer total message
    totalMessage: 'total',

    // Footer selected message
    selectedMessage: 'selected'
  };
